components/email_aliases/email_aliases_service.cc

```diff
@@ -318,4 +385,155 @@ std::string EmailAliasesService::GetAuthTokenForTesting() const {
+             std::unique_ptr<network::SimpleURLLoader> /*keep_alive_loader*/,
+             std::optional<std::string> response_body) {
+            std::move(inner_callback).Run(std::move(response_body));
+          },
+          std::move(callback), std::move(simple_url_loader)),
+      kMaxResponseLength);
+}
+
+void EmailAliasesService::OnGenerateAliasResponse(
+    GenerateAliasCallback user_callback,
+    std::optional<std::string> response_body) {
+  auto parsed = ParseResponseDictOrError(response_body);
+  if (!parsed.has_value()) {
+    std::move(user_callback).Run(base::unexpected(parsed.error()));
+    return;
+  }
+  if (auto ok = GenerateAliasResponse::FromValue(parsed.value())) {
+    if (ok->message == "created" && !ok->alias.empty()) {
+      std::move(user_callback).Run(base::ok(ok->alias));
+      return;
+    }
+  }
+  std::move(user_callback)
+      .Run(base::unexpected(l10n_util::GetStringUTF8(
+          IDS_EMAIL_ALIASES_ERROR_ALIAS_NOT_AVAILABLE)));
+}
+
+void EmailAliasesService::OnUpdateAliasResponse(
+    UpdateAliasCallback user_callback,
+    std::optional<std::string> response_body) {
+  RefreshAliases();
+  auto parsed = ParseResponseDictOrError(response_body);
+  if (!parsed.has_value()) {
+    std::move(user_callback).Run(base::unexpected(parsed.error()));
+    return;
+  }
+  if (auto edited = AliasEditedResponse::FromValue(parsed.value())) {
+    if (edited->message != "updated") {
+      std::move(user_callback).Run(base::unexpected(edited->message));
+      return;
+    }
+  }
+  std::move(user_callback).Run(base::ok(std::monostate{}));
+}
+
+void EmailAliasesService::OnDeleteAliasResponse(
+    DeleteAliasCallback user_callback,
+    std::optional<std::string> response_body) {
+  RefreshAliases();
+  auto parsed = ParseResponseDictOrError(response_body);
+  if (!parsed.has_value()) {
+    std::move(user_callback).Run(base::unexpected(parsed.error()));
+    return;
+  }
+  if (auto edited = AliasEditedResponse::FromValue(parsed.value())) {
+    if (edited->message != "deleted") {
+      std::move(user_callback).Run(base::unexpected(edited->message));
+      return;
+    }
+  }
+  std::move(user_callback).Run(base::ok(std::monostate{}));
+}
+
+void EmailAliasesService::RefreshAliases() {
+  ApiFetch(email_aliases_service_base_url_.Resolve("?status=active"),
+           net::HttpRequestHeaders::kGetMethod,
+           base::BindOnce(&EmailAliasesService::OnRefreshAliasesResponse,
+                          weak_factory_.GetWeakPtr()));
+}
+
+void EmailAliasesService::OnRefreshAliasesResponse(
+    std::optional<std::string> response_body) {
+  // TODO(https://github.com/brave/brave-browser/issues/48959):
+  // In this function, when an error happens, we should show
+  // an error message to the user (requires design work)
+  if (!response_body) {
+    LOG(ERROR) << "Email Aliases service error: No response body";
+    return;
+  }
+  auto parsed = base::JSONReader::Read(*response_body);
```

Comment:
Is there a reason why we are parsing this manually, rather than having an IDL type for it. It looks like it is just

```
dictionary Foo {
  AliasListEntry messages[];
};
```

Reply:
Unfortunately it doesn't appear there's a way to create a standalone list type?

Reply:
Fixed.

I added an AliasListResponse type. While the server API still responds with the old JSON list, the code will wrap the list in a dictionary before parsing. In the future, when we update the server API (in https://github.com/brave-intl/email-alias/issues/115) then we can remove the wrapping code.